solidity/contracts/l2/L2TBTC.sol

```diff
@@ -0,0 +1,81 @@
+// SPDX-License-Identifier: GPL-3.0-only
+
+// ██████████████     ▐████▌     ██████████████
+// ██████████████     ▐████▌     ██████████████
+//               ▐████▌    ▐████▌
+//               ▐████▌    ▐████▌
+// ██████████████     ▐████▌     ██████████████
+// ██████████████     ▐████▌     ██████████████
+//               ▐████▌    ▐████▌
+//               ▐████▌    ▐████▌
+//               ▐████▌    ▐████▌
+//               ▐████▌    ▐████▌
+//               ▐████▌    ▐████▌
+//               ▐████▌    ▐████▌
+
+pragma solidity ^0.8.17;
+
+import "@openzeppelin/contracts-upgradeable/token/ERC20/ERC20Upgradeable.sol";
+import "@openzeppelin/contracts-upgradeable/access/OwnableUpgradeable.sol";
+
+// TODO:
+// * Be paused by any one of `n` guardians, allowing avoidance of contagion in case
+//  of a chain- or bridge-specific incident.
```

Comment:
@mhluongo Would you like to pause everything or just mints/burns. I assumed mint/burns given the bridges are posing most of the risk. It is not about blocking l2 DEXes, for example.

Reply:
Just mints and burns

Reply:
What about `ERC20WithPermit, MisfundRecovery` functionality for tokens that will be inheriting from `L2TBTC.sol`? Do we want/need to add it?

Reply:
We want it! See c710eecaa7b2131374258801c3fa2e6c0b6fc118.